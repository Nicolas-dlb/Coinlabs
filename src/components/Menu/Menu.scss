@import "theme/_variables.scss";
.toggle_menu {
  transform: none !important;
}
.is_active {
  .menu_item_icon {
    background: linear-gradient(
      146deg,
      rgba(11, 171, 80, 1) 0%,
      rgba(30, 176, 149, 1) 100%
    ) !important;
  }

  color: var(--primary-font-color) !important;
}

.menu {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  width: fit-content;
  position: absolute;
  top: 90px;
  left: 0;
  bottom: 0;
  margin-left: 5vw;
  z-index: 10000;
  transform: translateX(-50vw);
  transition: all 0.5s ease;

  background-color: var(--app-bg);

  .user {
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: flex-start;
    margin-bottom: 4vh;
    height: 18vh;

    .user_icon {
      border-radius: 50%;
      width: 45px;
      height: 45px;
      margin-bottom: 1vh;
    }
    .user_name {
      font-size: 20px;
      margin-top: 1vh;
    }
    .user_email {
      font-size: 14px;
      color: var(--secondary-font-color);
    }
  }
  nav {
    a {
      text-decoration: none;
    }
  }
  .john {
    display: none;
  }
}
@media only screen and (min-width: 992px) {
  .menu {
    transform: none;
    margin-left: 0;
    padding-left: 30px;
    padding-right: 40px;
    background-color: var(--secondary-bg-color);
    .user {
      display: none;
    }
    .menu_changelog {
      display: none !important;
    }
    .menu_item {
      margin: 30px 0 !important;
    }
    .john {
      display: inline-block;
      position: absolute;
      bottom: 5vh;
      right: 1vw;
      text-align: center;
      img {
        width: 100%;
      }
      p {
        font-size: 15px;
        margin-bottom: 5px;
      }
      p:nth-child(3) {
        color: #848093;
        font-size: 13px;
        margin-bottom: 15px;
      }
      span {
        cursor: pointer;
        border-radius: 9px;
        padding: 10px 45px;
        background: rgba(11, 171, 80, 1);
      }
    }
  }
}
