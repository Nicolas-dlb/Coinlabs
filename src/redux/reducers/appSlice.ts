import { createSlice } from "@reduxjs/toolkit";

const initialState = {
  search: "Bitcoin",
};

const appSlice = createSlice({
  name: "app",
  initialState,
  reducers: {
    setSearch: (state, action) => {
      state.search = action.payload;
    },
  },
});
export const selectSearch = (state: any) => state.app.search;
export const { setSearch } = appSlice.actions;
export default appSlice.reducer;
