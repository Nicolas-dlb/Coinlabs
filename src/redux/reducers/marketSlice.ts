import { createSlice } from "@reduxjs/toolkit";

type CryptoState = {
  market: object;
  history: any;
  variation: object;
  time: string;
};
const initialState: CryptoState = {
  market: {},
  history: {},
  variation: { ethereum: 0, bitcoin: 0, ripple: 0, litecoin: 0, neo: 0 },
  time: "month",
};

const marketSlice = createSlice({
  name: "market",
  initialState,
  reducers: {
    loadCrypto: (state: CryptoState, action: any) => {
      state.market = { ...action.payload };
    },
    loadHistory: (state: CryptoState, action: any) => {
      state.history = { ...state.history, ...action.payload };
    },
    setVariation: (state: CryptoState, action: any) => {
      state.variation = { ...state.variation, ...action.payload };
    },
    setTime: (state: CryptoState, action: any) => {
      state.time = action.payload;
    },
  },
});

export const selectMarket = (state: any) => state.market.market;
export const selectHistory = (state: any) => state.market.history;
export const selectVariation = (state: any) => state.market.variation;
export const selectTime = (state: any) => state.market.time;
export const { loadCrypto, loadHistory, setVariation, setTime } =
  marketSlice.actions;
export default marketSlice.reducer;
