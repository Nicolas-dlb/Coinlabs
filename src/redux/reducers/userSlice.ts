import { createSlice } from "@reduxjs/toolkit";

const initialState = {
  userName: null,
  userEmail: null,
  profilPic: null,
};

const userSlice = createSlice({
  name: "user",
  initialState,
  reducers: {
    login: (state, action) => {
      state.userName = action.payload.userName;
      state.userEmail = action.payload.email;
    },
    logout: (state) => {
      state.userName = null;
      state.userEmail = null;
    },
    setUsername: (state, action) => {
      state.userName = action.payload;
    },
    setProfilPic: (state, action) => {
      state.profilPic = action.payload;
    },
    resetProfilPic: (state) => {
      state.profilPic = null;
    },
  },
});
export const selectUserName = (state: any) => state.user.userName;
export const selectUserEmail = (state: any) => state.user.userEmail;
export const selectProfilPic = (state: any) => state.user.profilPic;
export const { login, logout, setUsername, setProfilPic, resetProfilPic } =
  userSlice.actions;
export default userSlice.reducer;
