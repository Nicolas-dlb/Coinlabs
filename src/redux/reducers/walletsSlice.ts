import { createSlice } from "@reduxjs/toolkit";

type WalletsState = {
  wallets: any;
  balanceHistory: any;
  walletHistory: any;
  cryptoHistory: any;
  incomeHistory: any;
  expensesHistory: any;
};
const initialState: WalletsState = {
  wallets: {
    usd: 20000,
    eur: 0,
    gbp: 0,
    mxn: 0,
    ethereum: 0,
    ethereumPrice: 0,
    neo: 0,
    neoPrice: 0,
    litecoin: 0,
    litecoinPrice: 0,
    bitcoin: 0,
    bitcoinPrice: 0,
    ripple: 0,
    ripplePrice: 0,
    TotalCrypto: 0,
    total: 20000,
  },
  balanceHistory: [],
  walletHistory: [],
  cryptoHistory: [],
  incomeHistory: [],
  expensesHistory: [],
};

function isIterable(value: any) {
  return Symbol.iterator in Object(value);
}

const walletsSlice = createSlice({
  name: "wallets",
  initialState,
  reducers: {
    loadWallets: (state: WalletsState, action: any) => {
      state.wallets = {
        ...state.wallets,
        ...action.payload,
        eur: state.wallets.usd * 0.82366545,
        gbp: state.wallets.usd * 0.710764,
        mxn: state.wallets.usd * 19.945724,
      };
    },
    setWallets: (state: WalletsState, action: any) => {
      state.wallets = action.payload.usd
        ? {
            ...state.wallets,
            ...action.payload,

            eur: action.payload.usd * 0.82366545,
            gbp: action.payload.usd * 0.710764,
            mxn: action.payload.usd * 19.945724,
          }
        : {
            ...state.wallets,
            ...action.payload,
          };
    },
    setBalanceHistory: (state: WalletsState, action: any) => {
      if (isIterable(state.balanceHistory)) {
        state.balanceHistory = [...state.balanceHistory, action.payload];
      } else {
        state.balanceHistory = [action.payload];
      }
    },
    loadBalanceHistory: (state: WalletsState, action: any) => {
      state.balanceHistory = action.payload;
    },
    resetBalanceHistory: (state: WalletsState) => {
      state.balanceHistory = [];
    },
    setWalletHistory: (state: WalletsState, action: any) => {
      if (isIterable(state.walletHistory)) {
        state.walletHistory = [action.payload, ...state.walletHistory];
      } else {
        state.walletHistory = [action.payload];
      }
    },
    resetWalletHistory: (state: WalletsState) => {
      state.walletHistory = [];
    },
    loadWalletHistory: (state: WalletsState, action: any) => {
      state.walletHistory = action.payload;
    },
    setCryptoHistory: (state: WalletsState, action: any) => {
      if (isIterable(state.cryptoHistory)) {
        state.cryptoHistory = [...state.cryptoHistory, action.payload];
      } else {
        state.cryptoHistory = [action.payload];
      }
    },
    loadCryptoHistory: (state: WalletsState, action: any) => {
      state.cryptoHistory = action.payload;
    },
    resetCryptoHistory: (state: WalletsState) => {
      state.cryptoHistory = [];
    },
    setIncomeHistory: (state: WalletsState, action: any) => {
      if (isIterable(state.incomeHistory)) {
        state.incomeHistory = [...state.incomeHistory, action.payload];
      } else {
        state.incomeHistory = [action.payload];
      }
    },
    loadIncomeHistory: (state: WalletsState, action: any) => {
      state.incomeHistory = action.payload;
    },
    resetIncomeHistory: (state: WalletsState) => {
      state.incomeHistory = [];
    },
    setExpensesHistory: (state: WalletsState, action: any) => {
      if (isIterable(state.expensesHistory)) {
        state.expensesHistory = [...state.expensesHistory, action.payload];
      } else {
        state.expensesHistory = [action.payload];
      }
    },
    loadExpensesHistory: (state: WalletsState, action: any) => {
      state.expensesHistory = action.payload;
    },
    resetExpensesHistory: (state: WalletsState) => {
      state.expensesHistory = [];
    },
    
  },
});

export const selectWallets = (state: any) => state.wallets.wallets;
export const selectBalanceHistory = (state: any) =>
  state.wallets.balanceHistory;
  export const selectWalletHistory = (state: any) => state.wallets.walletHistory;
export const selectCryptoHistory = (state: any) => state.wallets.cryptoHistory;
export const selectIncomeHistory = (state: any) => state.wallets.incomeHistory;
export const selectExpensesHistory = (state: any) => state.wallets.expensesHistory;
export const {
  loadWallets,
  setWallets,
  setBalanceHistory,
  loadBalanceHistory,
  resetBalanceHistory,
  setWalletHistory,
  resetWalletHistory,
  loadWalletHistory,
  setIncomeHistory,
  resetIncomeHistory,
  loadIncomeHistory,
  setExpensesHistory,
  resetExpensesHistory,
  loadExpensesHistory,
  setCryptoHistory,
  resetCryptoHistory,
  loadCryptoHistory,
} = walletsSlice.actions;
export default walletsSlice.reducer;
