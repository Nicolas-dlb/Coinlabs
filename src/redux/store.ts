import { configureStore } from "@reduxjs/toolkit";
import marketReducer from "./reducers/marketSlice";
import walletsReducer from "./reducers/walletsSlice";
import appReducer from "./reducers/appSlice";
import userReducer from "./reducers/userSlice";

export default configureStore({
  reducer: {
    market: marketReducer,
    wallets: walletsReducer,
    app: appReducer,
    user: userReducer,
  },
});
