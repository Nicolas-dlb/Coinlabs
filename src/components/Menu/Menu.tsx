import './Menu.scss';
import market from 'assets/pictures/Market.png';
import { useState } from 'react';
import john from 'assets/pictures/John.svg';
import { useSelector } from 'react-redux';
import { selectUserEmail, selectUserName } from 'redux/reducers/userSlice';
import { Link } from "react-router-dom";
function Menu() {
    const [active, setActive] = useState("dashboard");
    const username = useSelector(selectUserName);
    const email = useSelector(selectUserEmail);
    
    return (
        <div id="menu" className="menu">
            <div className="user">
                
                     <img className="user_icon" src="https://scontent-cdt1-1.xx.fbcdn.net/v/t1.6435-9/119068627_10214566640554345_3311555602242066403_n.jpg?_nc_cat=103&ccb=1-3&_nc_sid=09cbfe&_nc_ohc=TD0JJun6_n0AX-RSVVr&_nc_ht=scontent-cdt1-1.xx&oh=6b73a5f5d399a29725277e3263783179&oe=60CC92DE" alt="" />
                
                <div className="user_name">
                    <p>{username}</p>
                </div>
                <div className="user_email">
                    <p>{email}</p>
                </div>
            </div>
           <nav>
               <Link to="/Dashboard">
               <div onClick={() => setActive("dashboard")} className={ active === "dashboard" ? "menu_item is_active" : "menu_item"}>
                   <div className="menu_item_icon">
                      <div className="dashboard_icon">
                        <div className="dashboard_icon_top">
                               <span className="square"></span>
                       <span className="square" style={{opacity: 0.6}}></span>
                        </div>
                      <div className="dashboard_icon_bottom">
                           <span className="square"></span>
                       <span className="square"></span>
                      </div>
                      </div>
                   </div>
                   <p>Dashboard</p>
               </div>
               </Link>
               
               <Link to="/Market" >
<div onClick={() => setActive("market")} className={ active === "market" ? "menu_item is_active" : "menu_item"}>
                   <div className="menu_item_icon">
                     <span className="fix_market_icon"></span>
                      <img className="market_icon" src={market} alt="" />
                   </div>
                   <p>Market</p>
               </div>
               </Link>
                
               <div onClick={() => setActive("portfolio")} className={ active === "portfolio" ? "menu_item is_active" : "menu_item"}>
                   <div className="menu_item_icon">
                     <svg xmlns="http://www.w3.org/2000/svg" width="512" height="512" viewBox="0 0 511.4 511.4"><g xmlns="http://www.w3.org/2000/svg"><path d="m0.7 252v-160.3c0-34.5 28-62.4 62.4-62.4h385c34.5 0 62.4 28 62.4 62.4v259.3c0 34.5-28 62.4-62.4 62.4h-122.5c-21 0-40.7-10.6-52.2-28.2l-28-42.6c-11.5-17.6-31.2-28.2-52.2-28.2h-130.1c-34.5-62.4-28-62.4-62.4z" data-original="#df8621" fill="#e4e4e4"/><path d="m415.2 0h-214.3c-26.5 0-48 21.5-48 48v284.4c0 26.5 21.5 48 48 48h214.3c26.5 0 48-21.5 48-48v-284.4c0-26.5-21.5-48-48-48z" data-original="#dddaec" fill="#8f8d97"/><path d="m338.6 73.2h-214.3c-26.5 0-48 21.5-48 48v284.4c0 26.5 21.5 48 48 48h214.3c26.5 0 48-21.5 48-48v-284.4c0-26.5-21.5-48-48-48z" data-original="#f1f4f6" fill="#f1f4f6"/><path d="m510.6 288.6v160.3c0 34.5-28 62.4-62.4 62.4h-385c-34.5 0-62.4-28-62.4-62.4v-259.3c0-34.5 28-62.4 62.4-62.4h122.5c21 0 40.7 10.6 52.2 28.2l28 42.6c11.5 17.6 31.2 28.2 52.2 28.2h130.1c34.5 0 62.4 28 62.4 62.4z" data-original="#e7a52e" className="a"/><path d="m114 511.3h-50.8c-34.5 0-62.4-28-62.4-62.4v-259.2c0-34.5 28-62.4 62.4-62.4h50.8c-34.5 0-62.4 28-62.4 62.4v259.2c0 34.5 28 62.4 62.4 62.4z" data-original="#e49824" className="a"/><path d="m316.8 198.4c-2.3 0-4.5-1.1-6-2.9l-26.9-33.4c-2.7-3.3-2.1-8.2 1.2-10.8s8.2-2.1 10.8 1.2l19.5 24.3 28.3-53.2c2-3.8 6.7-5.2 10.4-3.2 3.8 2 5.2 6.7 3.2 10.4l-33.8 63.5c-1.3 2.5-4 4.1-6.8 4.1z" data-original="#b3e59f" fill="#f2f2f2"/><g fill="#ffe07d"><path d="m191.9 281.6h-131.6c-4.3 0-7.7-3.5-7.7-7.7s3.5-7.7 7.7-7.7h131.6c4.3 0 7.7 3.5 7.7 7.7 0 4.3-3.5 7.7-7.7 7.7z" data-original="#ffe07d" className="b"/><path d="m191.9 310.1h-131.6c-4.3 0-7.7-3.5-7.7-7.7s3.5-7.7 7.7-7.7h131.6c4.3 0 7.7 3.5 7.7 7.7 0 4.3-3.5 7.7-7.7 7.7z" data-original="#ffe07d" className="b"/></g></g></svg>
                   </div>
                   <p>Portfolio</p>
               </div>
                <div onClick={() => setActive("history")} className={ active === "history" ? "menu_item is_active" : "menu_item"}>
                   <div onClick={() => setActive("history")} className="menu_item_icon">
                     <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 48 48" width="48" height="48"><circle cx="24" cy="24" r="18.5" style={{fill: "white",strokeWidth: "0", stroke: "#000"}}/><polyline points="22.5 13.5 22.5 24.5 27.5 29.5" style={{fill: "none",strokeLinejoin:"round",strokeWidth:"3",stroke:"#000"}}/></svg>
                   </div>
                   <p>History</p>
               </div>
               <div onClick={() => setActive("settings")} className={ active === "settings" ? "menu_item is_active" : "menu_item"}>
                   <div onClick={() => setActive("settings")} className="menu_item_icon">
                     <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path d="M496.6 312.1l-47.1-36.8c1.5-12.8 1.5-25.8 0-38.7l47.1-36.8c8.8-7.1 11.2-19.6 5.6-29.4l-49-84.6c-5.6-9.9-17.6-14.2-28.3-10.2l-55.5 22.3c-10.4-7.6-21.6-14-33.3-19.3l-8.5-58.9c-1.5-11.3-11.2-19.7-22.7-19.6h-98.1c-11.3-0.1-20.9 8.2-22.5 19.5l-8.5 59.1c-11.7 5.4-22.8 11.8-33.3 19.4L86.9 75.6c-10.5-4.2-22.5 0.1-28.1 10L9.8 170.3c-5.8 9.9-3.4 22.5 5.6 29.5l47.1 36.8c-1.5 12.8-1.5 25.8 0 38.7l-47.1 36.8c-8.8 7.1-11.2 19.6-5.6 29.4l48.9 84.7c5.6 9.9 17.6 14.2 28.3 10.2l55.5-22.3c10.4 7.6 21.6 14.1 33.3 19.4l8.5 58.9c1.5 11.3 11.1 19.7 22.5 19.6h98.1c11.3 0.1 21-8.2 22.6-19.5l8.5-59.1c11.7-5.4 22.8-11.8 33.3-19.4l55.7 22.4c10.6 4.1 22.5-0.1 28.1-10l49.2-85.1C507.9 331.5 505.4 319.1 496.6 312.1zM256 362.7c-58.9 0-106.7-47.8-106.7-106.7s47.8-106.7 106.7-106.7 106.7 47.8 106.7 106.7C362.6 314.9 314.8 362.6 256 362.7z" fill="#fff"/></svg>
                   </div>
                   <p>Settings</p>
               </div>
           </nav>
           <div className="john">
               <img src={john} alt="" />
               <p>Newest version soon available!</p>
               <p>More info version 1.2</p>
               <span>Update</span>
           </div>
        </div>
    )
}

 

export default Menu
