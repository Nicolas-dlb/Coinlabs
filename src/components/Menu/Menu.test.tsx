import React from "react";
import { mount } from "enzyme";
import store from "redux/store";
import { Provider } from "react-redux";
import { login, setProfilPic, setUsername } from "redux/reducers/userSlice";
import Menu from "./Menu";

describe("Menu", () => {
  let MenuComponent: any;
  beforeEach(async () => {
    MenuComponent = mount(
      <Provider store={store}>
        <Menu />
      </Provider>
    );
  });
  afterEach(() => {
    MenuComponent = null;
  });

  it("should render correctly", () => {
    expect(MenuComponent).toMatchSnapshot();
  });

  it("should render 5 MenuItem components", () => {
    expect(MenuComponent.find(".menu_item").length).toBe(5);
  });

  it("should render user profil picture", () => {
    store.dispatch(
      setProfilPic(
        "https://www.vhv.rs/dpng/d/164-1645859_selfie-clipart-groucho-glass-good-profile-hd-png.png"
      )
    );
    expect(MenuComponent.find(".user_icon").props().src).toBe(
      store.getState().user.profilPic
    );
  });

  it("should render user name name", () => {
    store.dispatch(setUsername("John Doe"));
    expect(MenuComponent.find(".user_name").text()).toBe(
      store.getState().user.userName
    );
  });

  it("should render user email email", () => {
    store.dispatch(login({ userName: "John Doe", email: "johndoe@gmail.com" }));
    expect(MenuComponent.find(".user_email").text()).toBe(
      store.getState().user.userEmail
    );
  });
});
