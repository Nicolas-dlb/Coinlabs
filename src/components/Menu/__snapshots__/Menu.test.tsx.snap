// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Menu>
    <div
      className="menu"
      id="menu"
    >
      <div
        className="user"
      >
        <img
          alt=""
          className="user_icon"
          src="https://www.vhv.rs/dpng/d/164-1645859_selfie-clipart-groucho-glass-good-profile-hd-png.png"
        />
        <div
          className="user_name"
        >
          <p />
        </div>
        <div
          className="user_email"
        >
          <p />
        </div>
      </div>
      <nav>
        <MenuItem
          active="dashboard"
          item="Dashboard"
          onClick={[Function]}
        >
          <div
            className="menu_item is_active menu_dashboard"
            id="menu_dashboard"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="menu_item_icon"
            >
              <div
                className="dashboard_icon"
              >
                <div
                  className="dashboard_icon_top"
                >
                  <span
                    className="square"
                  />
                  <span
                    className="square"
                    style={
                      Object {
                        "opacity": 0.6,
                      }
                    }
                  />
                </div>
                <div
                  className="dashboard_icon_bottom"
                >
                  <span
                    className="square"
                  />
                  <span
                    className="square"
                  />
                </div>
              </div>
            </div>
            <p>
              Dashboard
            </p>
          </div>
        </MenuItem>
        <MenuItem
          active="dashboard"
          item="Market"
          onClick={[Function]}
        >
          <div
            className="menu_item menu_market"
            id="menu_market"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="menu_item_icon"
            >
              <span
                className="fix_market_icon"
              />
              <img
                alt=""
                className="market_icon"
                src={
                  Object {
                    "canInstrument": true,
                    "createTransformer": [Function],
                    "getCacheKey": [Function],
                    "process": [Function],
                  }
                }
              />
            </div>
            <p>
              Market
            </p>
          </div>
        </MenuItem>
        <MenuItem
          active="dashboard"
          item="Portfolio"
          onClick={[Function]}
        >
          <div
            className="menu_item menu_portfolio"
            id="menu_portfolio"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="menu_item_icon"
            >
              <svg
                height="512"
                viewBox="0 0 511.4 511.4"
                width="512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m0.7 252v-160.3c0-34.5 28-62.4 62.4-62.4h385c34.5 0 62.4 28 62.4 62.4v259.3c0 34.5-28 62.4-62.4 62.4h-122.5c-21 0-40.7-10.6-52.2-28.2l-28-42.6c-11.5-17.6-31.2-28.2-52.2-28.2h-130.1c-34.5-62.4-28-62.4-62.4z"
                    data-original="#df8621"
                    fill="#e4e4e4"
                  />
                  <path
                    d="m415.2 0h-214.3c-26.5 0-48 21.5-48 48v284.4c0 26.5 21.5 48 48 48h214.3c26.5 0 48-21.5 48-48v-284.4c0-26.5-21.5-48-48-48z"
                    data-original="#dddaec"
                    fill="#8f8d97"
                  />
                  <path
                    d="m338.6 73.2h-214.3c-26.5 0-48 21.5-48 48v284.4c0 26.5 21.5 48 48 48h214.3c26.5 0 48-21.5 48-48v-284.4c0-26.5-21.5-48-48-48z"
                    data-original="#f1f4f6"
                    fill="#f1f4f6"
                  />
                  <path
                    className="a"
                    d="m510.6 288.6v160.3c0 34.5-28 62.4-62.4 62.4h-385c-34.5 0-62.4-28-62.4-62.4v-259.3c0-34.5 28-62.4 62.4-62.4h122.5c21 0 40.7 10.6 52.2 28.2l28 42.6c11.5 17.6 31.2 28.2 52.2 28.2h130.1c34.5 0 62.4 28 62.4 62.4z"
                    data-original="#e7a52e"
                  />
                  <path
                    className="a"
                    d="m114 511.3h-50.8c-34.5 0-62.4-28-62.4-62.4v-259.2c0-34.5 28-62.4 62.4-62.4h50.8c-34.5 0-62.4 28-62.4 62.4v259.2c0 34.5 28 62.4 62.4 62.4z"
                    data-original="#e49824"
                  />
                  <path
                    d="m316.8 198.4c-2.3 0-4.5-1.1-6-2.9l-26.9-33.4c-2.7-3.3-2.1-8.2 1.2-10.8s8.2-2.1 10.8 1.2l19.5 24.3 28.3-53.2c2-3.8 6.7-5.2 10.4-3.2 3.8 2 5.2 6.7 3.2 10.4l-33.8 63.5c-1.3 2.5-4 4.1-6.8 4.1z"
                    data-original="#b3e59f"
                    fill="#f2f2f2"
                  />
                  <g
                    fill="#ffe07d"
                  >
                    <path
                      className="b"
                      d="m191.9 281.6h-131.6c-4.3 0-7.7-3.5-7.7-7.7s3.5-7.7 7.7-7.7h131.6c4.3 0 7.7 3.5 7.7 7.7 0 4.3-3.5 7.7-7.7 7.7z"
                      data-original="#ffe07d"
                    />
                    <path
                      className="b"
                      d="m191.9 310.1h-131.6c-4.3 0-7.7-3.5-7.7-7.7s3.5-7.7 7.7-7.7h131.6c4.3 0 7.7 3.5 7.7 7.7 0 4.3-3.5 7.7-7.7 7.7z"
                      data-original="#ffe07d"
                    />
                  </g>
                </g>
              </svg>
            </div>
            <p>
              Portfolio
            </p>
          </div>
        </MenuItem>
        <MenuItem
          active="dashboard"
          item="Changelog"
          onClick={[Function]}
        >
          <div
            className="menu_item menu_changelog"
            id="menu_changelog"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="menu_item_icon"
            >
              <svg
                height="512"
                viewBox="0 0 512 512"
                width="512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M456.3 130.9V44.5C456.3 19.9 436.4 0 411.8 0H100.2C75.6 0 55.7 19.9 55.7 44.5v219.9C24.7 265.9 0 291.5 0 322.8s24.7 56.9 55.7 58.4v86.3c0 24.6 19.9 44.5 44.5 44.5h311.7c24.6 0 44.5-19.9 44.5-44.5V247.6C487.3 246.1 512 220.5 512 189.2S487.3 132.3 456.3 130.9zM16.7 322.8c0-22.1 17.2-40.2 39-41.6v83.3C33.9 363 16.7 344.9 16.7 322.8zM456.3 230.9v-83.3c21.7 1.4 39 19.6 39 41.6S478.1 229.4 456.3 230.9z"
                  data-original="#f2efe2"
                  fill="#fff"
                  xmlns="http://www.w3.org/2000/svg"
                />
                <path
                  d="M91.8 139.1c0-4.6 3.7-8.3 8.3-8.3h100.2c4.6 0 8.3 3.7 8.3 8.3 0 4.6-3.7 8.3-8.3 8.3H100.2C95.6 147.5 91.8 143.7 91.8 139.1zM411.8 364.5H311.7c-4.6 0-8.3 3.7-8.3 8.3 0 4.6 3.7 8.3 8.3 8.3h100.2c4.6 0 8.3-3.7 8.3-8.3C420.2 368.3 416.4 364.5 411.8 364.5zM211.5 286.6H100.2c-4.6 0-8.3 3.7-8.3 8.3s3.7 8.3 8.3 8.3h111.3c4.6 0 8.3-3.7 8.3-8.3S216.1 286.6 211.5 286.6zM411.8 286.6H311.7c-4.6 0-8.3 3.7-8.3 8.3s3.7 8.3 8.3 8.3h100.2c4.6 0 8.3-3.7 8.3-8.3S416.4 286.6 411.8 286.6zM203.1 217c0 4.6 3.7 8.3 8.3 8.3h200.3c4.6 0 8.3-3.7 8.3-8.3 0-4.6-3.7-8.3-8.3-8.3H211.5C206.9 208.7 203.1 212.4 203.1 217zM100.2 225.4h44.5c4.6 0 8.3-3.7 8.3-8.3 0-4.6-3.7-8.3-8.3-8.3h-44.5c-4.6 0-8.3 3.7-8.3 8.3C91.8 221.7 95.6 225.4 100.2 225.4zM177 208.7c-4.6 0-8.3 3.7-8.3 8.3 0 4.6 3.7 8.3 8.3 8.3h1.1c4.6 0 8.3-3.7 8.3-8.3 0-4.6-3.7-8.3-8.3-8.3H177zM246 286.6h-1.1c-4.6 0-8.3 3.7-8.3 8.3s3.7 8.3 8.3 8.3h1.1c4.6 0 8.3-3.7 8.3-8.3S250.6 286.6 246 286.6zM278.3 286.6h-1.1c-4.6 0-8.3 3.7-8.3 8.3s3.7 8.3 8.3 8.3h1.1c4.6 0 8.3-3.7 8.3-8.3S282.9 286.6 278.3 286.6z"
                  data-original="#bfbba3"
                  fill="#878474"
                  xmlns="http://www.w3.org/2000/svg"
                />
                <path
                  d="M263.9 379.1l-43.3 34.2c-2.2 1.7-5.7 1.7-7.9 0 -1-0.8-1.6-1.9-1.6-3.1v-28.9H64h-5.6C26.2 381.2 0 355 0 322.8c0-31.3 24.7-56.9 55.7-58.4v16.7c-21.7 1.4-39 19.6-39 41.6 0 23 18.7 41.7 41.7 41.7H64h147.1v-28.9c0-1.2 0.6-2.3 1.6-3.1 2.2-1.7 5.7-1.7 7.9 0l43.3 34.2C268.2 370.1 268.2 375.7 263.9 379.1zM453.6 130.8H448 300.5v-28.9c0-1.2-0.6-2.3-1.6-3.1 -2.2-1.7-5.7-1.7-7.9 0l-43.3 34.2c-4.3 3.4-4.3 9 0 12.4l43.3 34.2c2.2 1.7 5.7 1.7 7.9 0 1-0.8 1.6-1.9 1.6-3.1v-28.9H448h5.6c23 0 41.7 18.7 41.7 41.7 0 22.1-17.2 40.2-39 41.6v16.7C487.3 246.1 512 220.5 512 189.2 512 157 485.8 130.8 453.6 130.8z"
                  data-original="#fc8059"
                  fill="#878474"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </svg>
            </div>
            <p>
              Changelog
            </p>
          </div>
        </MenuItem>
        <MenuItem
          active="dashboard"
          item="Settings"
          onClick={[Function]}
        >
          <div
            className="menu_item menu_settings"
            id="menu_settings"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="menu_item_icon"
            >
              <svg
                fill="white"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M22.2 14.4L21 13.7c-1.3-0.8-1.3-2.7 0-3.5l1.2-0.7c1-0.6 1.3-1.8 0.7-2.7l-1-1.7c-0.6-1-1.8-1.3-2.7-0.7L18 5.1c-1.3 0.8-3-0.2-3-1.7V2c0-1.1-0.9-2-2-2h-2C9.9 0 9 0.9 9 2v1.3c0 1.5-1.7 2.5-3 1.7L4.8 4.4c-1-0.6-2.2-0.2-2.7 0.7l-1 1.7C0.6 7.8 0.9 9 1.8 9.6L3 10.3C4.3 11 4.3 13 3 13.7l-1.2 0.7c-1 0.6-1.3 1.8-0.7 2.7l1 1.7c0.6 1 1.8 1.3 2.7 0.7L6 18.9c1.3-0.8 3 0.2 3 1.7V22c0 1.1 0.9 2 2 2h2c1.1 0 2-0.9 2-2v-1.3c0-1.5 1.7-2.5 3-1.7l1.2 0.7c1 0.6 2.2 0.2 2.7-0.7l1-1.7C23.4 16.2 23.1 15 22.2 14.4zM12 16c-2.2 0-4-1.8-4-4 0-2.2 1.8-4 4-4s4 1.8 4 4C16 14.2 14.2 16 12 16z"
                />
              </svg>
            </div>
            <p>
              Settings
            </p>
          </div>
        </MenuItem>
      </nav>
      <div
        className="john"
      >
        <img
          alt=""
          src={
            Object {
              "canInstrument": true,
              "createTransformer": [Function],
              "getCacheKey": [Function],
              "process": [Function],
            }
          }
        />
        <p>
          Newest version soon available!
        </p>
        <p>
          More info version 1.1
        </p>
        <span
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          Read
        </span>
      </div>
    </div>
  </Menu>
</Provider>
`;
